test/transcoding/atomics_1.2.ll

```diff
@@ -80,12 +80,12 @@ define spir_kernel void @test_atomic_global(i32 addrspace(1)* %dst) #0 !kernel_a
 ; Function Attrs: nounwind
 define spir_kernel void @test_atomic_local(i32 addrspace(3)* %dst) #0 !kernel_arg_addr_space !11 !kernel_arg_access_qual !2 !kernel_arg_type !3 !kernel_arg_base_type !5 !kernel_arg_type_qual !4 {
   ; atomic_inc
-  %inc_il = tail call spir_func i32 @_Z10atomic_incPVU3AS3i(i32 addrspace(3)* %dst) #0
-  ; CHECK: _Z25atomic_fetch_add_explicitPU3AS3VU7_Atomicii12memory_order12memory_scope(i32 addrspace(3)* {{.*}}, i32 1
+  %inc_il = tail call spir_func i32 @_Z10atomic_incPU3AS3Vi(i32 addrspace(3)* %dst) #0
+  ; CHECK: _Z10atomic_incPU3AS3Vi(i32 addrspace(3)* %dst) #0
 
   ; atomic dec
-  %dec_jl = tail call spir_func i32 @_Z10atomic_decPVU3AS3j(i32 addrspace(3)* %dst) #0
-  ; CHECK: _Z25atomic_fetch_sub_explicitPU3AS3VU7_Atomicii12memory_order12memory_scope(i32 addrspace(3)* {{.*}}, i32 1
+  %dec_jl = tail call spir_func i32 @_Z10atomic_decPU3AS3Vj(i32 addrspace(3)* %dst) #0
+  ; CHECK: _Z10atomic_decPU3AS3Vi(i32 addrspace(3)* %dst) #0
 
   ; atomic_max
   %max_il = tail call spir_func i32 @_Z10atomic_maxPVU3AS3ii(i32 addrspace(3)* %dst, i32 0) #0
```

Comment:
As a target, we also need to fix the translation of those builtins. This is a test for opencl 1.2 and we expect 2.0 version of builtin:
_Z25atomic_fetch_max_explicitPU3AS3VU7_Atomicii12memory_order12memory_scope
Additionally, this builtin is incorrect, because pointer address space has to be generic (4 not 3) to be consistent with clang.

I'll handle all the builtins below in the next PR, because it requires additional map specific for OpenCL 1.2.